provider/pkg/provider/provider.go

```diff
@@ -501,12 +505,31 @@ func (k *googleCloudProvider) Delete(ctx context.Context, req *rpc.DeleteRequest
 		return nil, errors.Errorf("resource '%s' not found", resourceKey)
 	}
 
-	if res.NoDelete {
-		// TODO: delete with Set (e.g. Policy resources)
+	uri := res.RelativePath(req.GetId())
+
+	if strings.HasSuffix(uri, ":getIamPolicy") {
+		uri = strings.ReplaceAll(uri, ":getIamPolicy", ":setIamPolicy")
+
+		resp, err := sendRequestWithTimeout(ctx, res.UpdateVerb, uri, map[string]interface{}{}, 0)
+		if err != nil {
+			return nil, fmt.Errorf("error sending request: %s", err)
+		}
+
+		_, err = k.waitForResourceOpCompletion(ctx, res.BaseUrl, resp)
+		if err != nil {
+			return nil, errors.Wrapf(err, "waiting for completion")
+		}
+
 		return &empty.Empty{}, nil
 	}
 
-	uri := res.RelativePath(req.GetId())
+	if res.NoDelete {
+		// At the time of writing, the classic GCP provider has the same behavior and warning for 10 resources.
+		glog.V(4).Infof("%q resources"+
```

Comment:
We seem to have a lot of NoDelete resources (270+). This warning makes sense regardless for noDelete resources, but we might be over estimating these types of resources.

Reply:
The majority of them (~200) are IamPolicies. I agree we may not actually want to generate some of those resources as resources.